libminifi/src/utils/Id.cpp

```diff
@@ -48,6 +48,8 @@ namespace nifi {
 namespace minifi {
 namespace utils {
 
+const char* Identifier::UUID_FORMAT_STRING = "%02hhx%02hhx%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx%02hhx%02hhx%02hhx%02hhx";
```

Comment:
would `constexpr` allow us to make it an in-class initializer?

Reply:
I think it only works in c++17 or maybe even c++14.

Reply:
you are right that from `c++17` `inline` comes to the rescue, but even in `c++11` we may (must) add an initializer to the declaration of a literal constexpr variable (and it is enough if it is not odr-used) (see [here](https://en.cppreference.com/w/cpp/language/static) in section `Constant static members`)

```
#include <stdio.h>

struct A{
  static constexpr const char* FORMAT = "%d\n";
};

int main(){
  // uncommenting the following will generate a linker error
  // auto& ref = A::FORMAT;
  printf(A::FORMAT, 5);
}
```
it is not pivotal to move it to the header, I just wanted to share that it is possible

Reply:
Please make sure I can't reassign the string to something else. `constexpr` or top-level `const` achieves that.

Reply:
Made `UUID_FORMAT_STRING` `constexpr`.